### DISCUSSION TITLE
Setting 'base' in configuration and build.format: ' file'causing header link to point at incorrect url.

### DISCUSSION THREAD

**Discussion by davidguo12345** [withastro/starlight#2347] [Q&A]
Hi, 

I'm experiencing an issue setting up docs with starlight where I need to set my base to a subpath (e.g. 'some/path) and also build.format to 'file'. This is however, is causing the homepage to be at 'some/path/index.html' but all the headers link instead to 'some/path.html' which is 404. It looks to be caused by line 49 in 'path.ts' where if path is falsy, it will return '/index.html' but if 'path' is truthy (such as when base has been set) it will only append '.html' and break the link. 

Is there a fix or workaround if this issue? 

Current doc structure:
```.
└── src/

    └── content/
    
        └── docs/

            ├── index.mdx (this is my home page)
            
            ├── folder1/
            
            │   ├── page1.md
            
            │   └── page2.md
            
            └── folder2/
            
                └── page3.md
 ```
                
Thanks!

**Comment by delucis** [comment]
Hi @davidguo12345 — thanks for raising this! This sounds like a bug to me. Could you make [a new issue](https://github.com/withastro/starlight/issues/new?assignees=&labels=&projects=&template=---01-bug-report.yml) with a reproduction so we can take a look? (Or if it seems like an easy fix, a PR is also an option, but it would be good to have a reproduction in any case to check the details of the behaviour.)

**Reply by davidguo12345** [reply]
Created the issue: https://github.com/withastro/starlight/issues/2383, Thanks!
